web_app/db/crud.py

```diff
@@ -362,23 +362,33 @@ def open_position(self, position_id: uuid.UUID, current_prices: dict) -> str | N
             logger.error(f"Position with ID {position_id} not found")
             return None
 
-    def get_total_amounts_for_open_positions(self) -> Decimal:
+    async def get_total_amounts_for_open_positions(self) -> dict:
         """
-        Calculates the total amount for all positions where status is 'OPENED'.
+        Calculates the total value for all open positions based on current token prices.
 
-        :return: Total amount for all opened positions
+        :return: A dictionary mapping token symbols to the total value of open positions in that token.
         """
         with self.Session() as db:
             try:
-                total_opened_amount = (
-                    db.query(func.sum(cast(Position.amount, Numeric)))
+                open_positions = (
+                    db.query(Position)
                     .filter(Position.status == Status.OPENED.value)
-                    .scalar()
+                    .all()
                 )
-                return total_opened_amount
+                current_prices = await self.get_current_prices()
+
+                total_amounts = {}
+                for position in open_positions:
```

Comment:
we want to have as a result only one value, why we have separated total_amount per token?

Reply:
so it should be total of all token_amounts in USDC